platform-sdk/swirlds-common/src/main/java/com/swirlds/common/system/BasicSoftwareVersion.java

```diff
@@ -108,29 +108,9 @@ public int compareTo(final SoftwareVersion that) {
         }
     }
 
-    /**
-     * {@inheritDoc}
-     */
-    @Override
-    public boolean equals(final Object o) {
-        if (this == o) {
-            return true;
-        }
-
-        if (!(o instanceof BasicSoftwareVersion that)) {
-            return false;
-        }
-
-        return softwareVersion == that.softwareVersion;
-    }
-
-    /**
-     * {@inheritDoc}
-     */
-    @Override
-    public int hashCode() {
-        return Long.hashCode(softwareVersion);
-    }
+    // Intentionally do not implement equals() or hashCode(). Although it is legal to do so, it is not required,
+    // and all platform operations should be functional without it. Since this class is used for platform testing,
+    // we should make sure this crutch is not available.
 
```

Comment:
as I understand it, the Hedera software doesn't implement this, so the test apps shouldn't either. Can you please make that more clear in this comment

Reply:
I feel like the current comment already says this directly.

> Since this class is used for platform testing,
    // we should make sure this crutch is not available.

Can you suggest a better wording?